dhis-2/dhis-test-integration/src/test/java/org/hisp/dhis/trackedentity/TrackedEntityServiceTest.java

```diff
@@ -335,29 +335,164 @@ void shouldOrderEntitiesByIdWhenParamCreatedAtSupplied() {
     TrackedEntityQueryParams params = new TrackedEntityQueryParams();
 
     params.setOrgUnits(Set.of(organisationUnit));
-    params.setOrders(List.of(new OrderParam("createdAt", SortDirection.ASC)));
+    params.setOrders(List.of(new OrderParam("created", SortDirection.ASC)));
+
+    List<Long> teiIdList = entityInstanceService.getTrackedEntityIds(params, true, true);
+
+    assertEquals(
+        List.of(
+            entityInstanceC1.getId(),
+            entityInstanceB1.getId(),
+            entityInstanceA1.getId(),
+            entityInstanceD1.getId()),
+        teiIdList);
+  }
+
+  @Test
+  void shouldOrderEntitiesByCreatedInDescOrder() {
+    injectSecurityContext(superUser);
+
+    entityInstanceA1.setCreated(DateTime.now().plusDays(1).toDate());
+    entityInstanceB1.setCreated(DateTime.now().toDate());
+    entityInstanceC1.setCreated(DateTime.now().minusDays(1).toDate());
+    entityInstanceD1.setCreated(DateTime.now().plusDays(2).toDate());
+    addEntityInstances();
+
+    TrackedEntityQueryParams params = new TrackedEntityQueryParams();
+
+    params.setOrgUnits(Set.of(organisationUnit));
+    params.setOrders(List.of(new OrderParam("created", SortDirection.DESC)));
 
     List<Long> teiIdList = entityInstanceService.getTrackedEntityIds(params, true, true);
 
     assertEquals(
         List.of(
+            entityInstanceD1.getId(),
```

Comment:
just a comment.
I think readability can be improved if we store ascOrdered list of tei ids and then use it in the assertion 
`assertEquals( ascOrdered, teiIdList )`